### ISSUE TITLE
IP-3844. Upgrade/downgrade replica set

### ISSUE THREAD

**Issue by idea2-pepito** [idea2-platform/pepito-test#1128]
We need to automate the process to perform an upgrade of servers in a replica set. This could be a common case for shared and dedicated databases.

In shared databases it could happen that they start growing and we need to increase the server capacity to hold them without having to start migration processes that will incur in downtime.

For dedicated databases this will be the path if they grow. We start with the smallest servers possible and we grow as we go.

So basically we should be able to select a replica set and indicate that we want to change servers to a new configuration. This process should start replacing node by node in the replica set until it is completed.
